sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVParameters.scala

```diff
@@ -71,6 +71,8 @@ private[sql] case class CSVParameters(parameters: Map[String, String]) extends L
 
   val nullValue = parameters.getOrElse("nullValue", "")
 
+  val codec = parameters.getOrElse("compression", parameters.getOrElse("codec", null))
```

Comment:
for this one i'd name the internally name compression or compressionCodec since codec can mean a lot of different things.